GitHubReferences.ps1

```diff
@@ -0,0 +1,241 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+function Get-GitHubReference
+{
+<#
+    .SYNOPSIS
+        Retrieve a reference of a given GitHub repository.
+
+    .DESCRIPTION
+        Retrieve a reference of a given GitHub repository.
+        The Git repo for this module can be found here: http://aka.ms/PowerShellForGitHub
+
+    .PARAMETER OwnerName
+        Owner of the repository.
+        If not supplied here, the DefaultOwnerName configuration property value will be used.
+
+    .PARAMETER RepositoryName
+        Name of the repository.
+        If not supplied here, the DefaultRepositoryName configuration property value will be used.
+
+    .PARAMETER Uri
+        Uri for the repository.
+        The OwnerName and RepositoryName will be extracted from here instead of needing to provide
+        them individually.
+
+    .PARAMETER Reference
+        Name of the reference, for example: "heads/<branch name>" for branches and "tags/<tag name>" for tags.
+        Gets all the references if not provided
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no commandline status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .OUTPUTS
+        [PSCustomObject]
+        Details of the git reference in the given repository
+
+    .EXAMPLE
+        Get-GitHubReference -OwnerName Powershell -RepositoryName PowerShellForGitHub -Reference heads/master
+#>
+    [CmdletBinding(
+        SupportsShouldProcess,
+        DefaultParametersetName='Elements')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSShouldProcess", "",
+    Justification="Methods called within here make use of PSShouldProcess, and the switch is passed on to them inherently.")]

```

Comment:
Moved this to the next line because this was too far away from the 100 column limit. I haven't seen this being done in other files though. Should I keep it the old way to make it consistent? Thoughts?